File: PP/Lista2/fib.scala
object fib {
    def main(args: Array[String]) : Unit = {
        def fib(x : Int) : BigInt =
            x match {
                case 0 => 0
                case 1 => 1
                case n => if n > 0 then fib (n - 1) + fib (n - 2) else throw new Exception("Ujemna liczba");
            }
        println(fib(8))
    }
}

File: PP/Lista2/fib.ml
(* O(2^n), bo wysokosc drzewa n
(każda kolejna warstwa ma 2*poprzednia)
int -> int *)
let rec fib n =
    match n with
    | 0 -> 0
    | 1 -> 1
    | n -> if n > 0 then fib (n - 1) + fib (n - 2) else failwith "Ujemna liczba";;

print_int (fib 12);;